stdlib/public/Backtracing/CoreSymbolication.swift

```diff
@@ -0,0 +1,360 @@
+                    UInt32, UInt32, CSNotificationBlock?) -> CSSymbolicatorRef =
+    symbol(coreSymbolicationHandle, "CSSymbolicatorCreateWithBinaryImageList")
+
+  static let CSSymbolicatorGetSymbolOwnerWithAddressAtTime:
+    @convention(c) (CSSymbolicatorRef, mach_vm_address_t,
+                    CSMachineTime) -> CSSymbolOwnerRef =
+    symbol(coreSymbolicationHandle, "CSSymbolicatorGetSymbolOwnerWithAddressAtTime")
+  static let CSSymbolicatorForeachSymbolOwnerAtTime:
+    @convention(c) (CSSymbolicatorRef, CSMachineTime, @convention(block) (CSSymbolOwnerRef) -> Void) -> UInt =
+      symbol(coreSymbolicationHandle, "CSSymbolicatorForeachSymbolOwnerAtTime")
+
+  // CSSymbolOwner
+  static let CSSymbolOwnerGetName:
+    @convention(c) (CSSymbolOwnerRef) -> UnsafePointer<CChar>? =
+    symbol(coreSymbolicationHandle, "CSSymbolOwnerGetName")
+  static let CSSymbolOwnerGetSymbolWithAddress:
+    @convention(c) (CSSymbolOwnerRef, mach_vm_address_t) -> CSSymbolRef =
+    symbol(coreSymbolicationHandle, "CSSymbolOwnerGetSymbolWithAddress")
+  static let CSSymbolOwnerGetSourceInfoWithAddress:
+    @convention(c) (CSSymbolOwnerRef, mach_vm_address_t) -> CSSourceInfoRef =
+    symbol(coreSymbolicationHandle, "CSSymbolOwnerGetSourceInfoWithAddress")
+  static let CSSymbolOwnerForEachStackFrameAtAddress:
+    @convention(c) (CSSymbolOwnerRef, mach_vm_address_t, CSStackFrameIterator) -> UInt =
+    symbol(coreSymbolicationHandle, "CSSymbolOwnerForEachStackFrameAtAddress")
+  static let CSSymbolOwnerGetBaseAddress:
+    @convention(c) (CSSymbolOwnerRef) -> mach_vm_address_t =
+    symbol(coreSymbolicationHandle, "CSSymbolOwnerGetBaseAddress")
+
+  // CSSymbol
+  static let CSSymbolGetRange:
+    @convention(c) (CSSymbolRef) -> CSRange =
+    symbol(coreSymbolicationHandle, "CSSymbolGetRange")
+  static let CSSymbolGetName:
+    @convention(c) (CSSymbolRef) -> UnsafePointer<CChar>? =
+    symbol(coreSymbolicationHandle, "CSSymbolGetName")
+  static let CSSymbolGetMangledName:
+    @convention(c) (CSSymbolRef) -> UnsafePointer<CChar>? =
+    symbol(coreSymbolicationHandle, "CSSymbolGetMangledName")
+
+  // CSSourceInfo
+  static let CSSourceInfoGetPath:
+    @convention(c) (CSSourceInfoRef) -> UnsafePointer<CChar>? =
+    symbol(coreSymbolicationHandle, "CSSourceInfoGetPath")
+  static let CSSourceInfoGetLineNumber:
+    @convention(c) (CSSourceInfoRef) -> UInt32 =
+    symbol(coreSymbolicationHandle, "CSSourceInfoGetLineNumber")
+  static let CSSourceInfoGetColumn:
+    @convention(c) (CSSourceInfoRef) -> UInt32 =
+    symbol(coreSymbolicationHandle, "CSSourceInfoGetColumn")
+}
+
+// .. Crash Reporter support ...................................................
+
+// We can't import swiftFoundation here, so there's no automatic bridging for
+// CFString.  As a result, we need to do the dance manually.
+
+private func toCFString(_ s: String) -> CFString! {
+  let bytes = Array<UInt8>(s.utf8)
```

Comment:
you can save an alloc + copy here if you write

```
var s = s
s.withUTF8 { ... }
```

then String will just vend you its internals (already utf8).

Reply:
(unless bridged from NSString in which case we'll alloc & copy)